src/components/CategoryPicker.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import Typography from '@material-ui/core/Typography';
+import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
+import Input from '@material-ui/core/Input';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import FormControl from '@material-ui/core/FormControl';
+import ListItemText from '@material-ui/core/ListItemText';
+import Select from '@material-ui/core/Select';
+import Checkbox from '@material-ui/core/Checkbox';
+
+export default function CategoryPicker(): JSX.Element {
+  const classes = useStyles();
+  const [category, setCategory] = React.useState<string[]>([]);
+
+  const handleChange = (event: React.ChangeEvent<{ value: unknown }>) =>
+    setCategory(event.target.value as string[]);
+
+  return (
+    <div>
+      <Typography component="h2" variant="h6" style={{ marginTop: 20 }}>
+        Restrict results to specific categories
+      </Typography>
+      <FormControl className={classes.formControl}>
+        <InputLabel id="demo-mutiple-checkbox-label">Select</InputLabel>
+        <Select
+          labelId="demo-mutiple-checkbox-label"
+          id="demo-mutiple-checkbox"
+          multiple
+          value={category}
+          onChange={handleChange}
+          input={<Input />}
+          renderValue={(selected) => (selected as string[]).join(', ')}
+          MenuProps={MenuProps}
+        >
+          {selector.map((name) => (
+            <MenuItem key={name} value={name}>
+              <Checkbox checked={category.indexOf(name) > -1} />
+              <ListItemText primary={name} />
+            </MenuItem>
+          ))}
+        </Select>
+      </FormControl>
+    </div>
+  );
+}
+
+const selector = [
+  'Birth, Baptism, and Christening',
+  'Marriage',
+  'Death',
+  'Census, Residence, and Lists',
+  'Immigration and Naturalization',
+  'Military',
+  'Probate',
+  'Other',
+];
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    formControl: {
+      margin: theme.spacing(1),
+      minWidth: 600,
+      maxWidth: 600,
+    },
+  })
+);
+
+const ITEM_HEIGHT = 48;
+const ITEM_PADDING_TOP = 8;
+const MenuProps = {
```

Comment:
Usually only classes and types start with uppercase. (And React function components, because React forces them to be.)

Reply:
Noted thanks, have changed to camel-case